### PR TITLE
Add DiffSynth blockwise ControlNet support to QwenImageControlNetModel

### LINKED ISSUE #12221: [Looking for community contribution] support DiffSynth Controlnet in diffusers
### Model/Pipeline/Scheduler description

Hi!
We want to add first party support for DiffSynth controlnet in diffusers, and we are looking for some help from the community! 

Let me know if you're interested! 

### Open source status

- [x] The model implementation is available.
- [x] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

https://huggingface.co/SahilCarterr/Qwen-Image-Blockwise-ControlNet-Canny
https://huggingface.co/SahilCarterr/Qwen-Image-Blockwise-ControlNet-Depth

### PR DESCRIPTION
**Pull request by aryabyte21**
## Summary

Adds support for the [DiffSynth-Studio blockwise ControlNet](https://github.com/modelscope/DiffSynth-Studio) architecture to the existing `QwenImageControlNetModel`, addressing [the maintainer's feedback on PR #12317](https://github.com/huggingface/diffusers/pull/12317#discussion_r1865041506) by integrating it into the existing class rather than creating a separate one.

### What changed

- Added `controlnet_block_type` config parameter to `QwenImageControlNetModel` (`"linear"` default for InstantX, `"blockwise"` for DiffSynth)
- Added `BlockWiseControlBlock` module: RMSNorm + MLP fusion block that normalizes both base hidden states and control features separately before fusing via Linear → GELU → Linear (zero-initialized output)
- No pipeline changes needed — both variants produce the same `controlnet_block_samples` output format
- Added conversion script for DiffSynth checkpoints (`scripts/convert_diffsynth_blockwise_controlnet_to_diffusers.py`)

### Key design decision

Per @yiyixuxu's review comment on #12317:
> "is this the only difference between Diffsynth & instantX controlnet, can we just add the change to the existing one?"

**Yes** — the only architectural difference is in the controlnet blocks themselves. The blockwise variant uses `BlockWiseControlBlock(x, y)` (two inputs: base + control features) instead of `zero_module(nn.Linear)(x)` (one input). Everything else — the pipeline, transformer integration, and output format — is identical.

### Available checkpoints

- [`DiffSynth-Studio/Qwen-Image-Blockwise-ControlNet-Canny`](https://huggingface.co/DiffSynth-Studio/Qwen-Image-Blockwise-ControlNet-Canny)
- [`DiffSynth-Studio/Qwen-Image-Blockwise-ControlNet-Depth`](https://huggingface.co/DiffSynth-Studio/Qwen-Image-Blockwise-ControlNet-Depth)
- [`DiffSynth-Studio/Qwen-Image-Blockwise-ControlNet-Inpaint`](https://huggingface.co/DiffSynth-Studio/Qwen-Image-Blockwise-ControlNet-Inpaint)

### Usage

```python
from diffusers import QwenImageControlNetModel

# Blockwise variant (DiffSynth)
controlnet = QwenImageControlNetModel.from_pretrained(
    "path/to/converted/blockwise-controlnet"
)

# Or create from base transformer
controlnet = QwenImageControlNetModel.from_transformer(
    transformer,
    controlnet_block_type="blockwise",
)
```

Closes #12221

## Test plan

- [x] Existing `test_qwen_controlnet` passes (no regression)
- [x] Existing `test_qwen_controlnet_multicondition` passes (no regression)
- [x] New `test_qwen_blockwise_controlnet` passes
- [x] New `test_qwen_blockwise_controlnet_from_transformer` passes
- [ ] Full GPU inference test with converted DiffSynth checkpoint (needs GPU)

### CODE REVIEW

**Comment by iwr-redmond** [comment]
cc @apolinario @sayakpaul
